modules/apps/forms-and-workflow/calendar/calendar-test/src/testIntegration/java/com/liferay/calendar/upgrade/v1_0_4/test/UpgradeClassNamesTest.java

```diff
@@ -222,6 +222,15 @@ protected void addAssetVocabulary(long classNameId) throws Exception {
 		AssetVocabularyLocalServiceUtil.addAssetVocabulary(_assetVocabulary);
 	}
 
+	protected ClassName addClassName(String value) {
+		ClassName className = ClassNameLocalServiceUtil.createClassName(
+			CounterLocalServiceUtil.increment());
+
+		className.setValue(value);
+
+		return ClassNameLocalServiceUtil.addClassName(className);
+	}
+
 	protected ResourcePermission addResourcePermission(
```

Comment:
Hey @SamZiemer, great work! Thanks!

I was getting problems  running tests twice. Apparently `ClassNameLocalService` has some kind of internal cache that was breaking tests running after the first one. It was quite weird, I guess it is very specific for class names. Anyway, I could circumvent this problem here.

Besides that, I did some small source formatting and little changes in other commits. As I've said before, these upgrade tests are somewhat complicated but always prove themselves useful in the worst time, so many thanks for your work again!